backend/downsample_test.py

```diff
@@ -0,0 +1,303 @@
+# Copyright 2020 Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+"""Test Module for downsample.py"""
+
+import os
+import random
+import tempfile
+
+import pytest
+
+from downsample import average_downsample
+from downsample import downsample
+from downsample import lttb_downsample
+from downsample import max_min_downsample
+from downsample import secondary_downsample
+from downsample import strategy_reducer
+from downsample import triangle_area
+from utils import convert_to_csv
+
+
+class TestDownsampleClass:
+    """Test cllass for downsample.py"""
+
+    @pytest.fixture
+    def records(self):
+        """Generates test data."""
+        return [
+            [1573149236256988, 100, 'PPX_ASYS'],
+            [1573149236257088, 100, 'PPX_ASYS'],
+            [1573149236257188, 300, 'PPX_ASYS'],
+            [1573149236257288, 100, 'PPX_ASYS'],
+            [1573149236257388, 100, 'PPX_ASYS'],
+            [1573149236257488, 100, 'PPX_ASYS'],
+            [1573149236257588, 100, 'PPX_ASYS'],
+            [1573149236257688, 100, 'PPX_ASYS'],
+            [1573149236257788, 5, 'PPX_ASYS'],
+            [1573149236257888, 100, 'PPX_ASYS']
+        ]
+
+    @pytest.fixture
+    def records_complex_case(self):
+        """Generates complex test data."""
+        start = 1573149236256988
+        records_complex_case = [[start, 100, 'SYS']]
+        for _ in range(500):
+            records_complex_case.append(
+                [
+                    records_complex_case[-1][0] + random.randint(1E4, 1E6),
+                    random.randint(0, 500),
+                    'SYS'
+                ])
+
+        last = -1
+        for record in records_complex_case:
+            assert record[0] >= last
+            last = record[0]
+
+        return records_complex_case
+
+    @pytest.mark.parametrize('max_records', [0, 5, 10, 20, 30, 40, 1000000])
+    def test_limit_of_records(self, records, max_records):
```

Comment:
A test method should focus on one method being tested. Looks this test method covered 4 methods in the class being tested. Better break it into 4 test methods.

Also, the best practice is that the Test names should include the condition being tested and the expected outcome.

A test's name should fully describe what a test does, including not only the state and details of the method being tested but also its expected outcome. For example, a method to transfer money probably contains several interesting behaviors, each of which should be tested individually. Each test should explicitly name not only the method being tested, but also the expected outcome and sometimes specific details of the state or input being tested.
e.g. 
test_transfer_money_deducts_from_source
test_transfer_money_adds_to_destination
test_transfer_money_insufficient_funds_throws_exception
test_multiply_two_negative_values_returns_positive_value
test_multiply_anything_with_nan_returns_nan

So, in your case, the test methods could be something like:
test_ max_min_downsample_max_strategy_records_length_limit
test_ max_min_downsample_min_strategy_records_length_limit
test_ average_downsample_records_length_limit
test_ lttb_downsample_records_length_limit